integration_test/third_party_apps_test/main_test.go

```diff
@@ -819,40 +820,53 @@ var defaultApps = map[string]bool{
 var gpuModels = map[string]accelerator{
 	// This is the A100 40G model; A100 80G is similar so skipping
 	"a100": {
-		model:         "a100",
-		fullName:      "nvidia-tesla-a100",
-		machineType:   "a2-highgpu-1g",
-		availableZone: "us-central1-a",
+		model:            "a100",
+		fullName:         "nvidia-tesla-a100",
+		machineType:      "a2-highgpu-1g",
+		availableZone:    "us-central1-a",
+		acceleratorCount: 1,
 	},
 	"v100": {
-		model:         "v100",
-		fullName:      "nvidia-tesla-v100",
-		machineType:   "n1-standard-2",
-		availableZone: "us-central1-a",
+		model:            "v100",
+		fullName:         "nvidia-tesla-v100",
+		machineType:      "n1-standard-2",
+		availableZone:    "us-central1-a",
+		acceleratorCount: 1,
 	},
 	"t4": {
-		model:         "t4",
-		fullName:      "nvidia-tesla-t4",
-		machineType:   "n1-standard-2",
-		availableZone: "us-central1-a",
+		model:            "t4",
+		fullName:         "nvidia-tesla-t4",
+		machineType:      "n1-standard-2",
+		availableZone:    "us-central1-a",
+		acceleratorCount: 1,
 	},
 	"p4": {
-		model:         "p4",
-		fullName:      "nvidia-tesla-p4",
-		machineType:   "n1-standard-2",
-		availableZone: "us-central1-a",
+		model:            "p4",
+		fullName:         "nvidia-tesla-p4",
+		machineType:      "n1-standard-2",
+		availableZone:    "us-central1-a",
+		acceleratorCount: 1,
 	},
 	"p100": {
-		model:         "p100",
-		fullName:      "nvidia-tesla-p100",
-		machineType:   "n1-standard-2",
-		availableZone: "us-central1-c",
+		model:            "p100",
+		fullName:         "nvidia-tesla-p100",
+		machineType:      "n1-standard-2",
+		availableZone:    "us-central1-c",
+		acceleratorCount: 1,
 	},
 	"l4": {
-		model:         "l4",
-		fullName:      "nvidia-l4",
-		machineType:   "g2-standard-4",
-		availableZone: "us-central1-a",
+		model:            "l4",
+		fullName:         "nvidia-l4",
+		machineType:      "g2-standard-4",
+		availableZone:    "us-central1-a",
+		acceleratorCount: 1,
+	},
+	"h100": {
+		model:            "h100",
+		fullName:         "nvidia-h100-80gb",
+		machineType:      "a3-highgpu-8g",
+		availableZone:    "us-west1-a",
+		acceleratorCount: 8,
```

Comment:
Can we parse this out of the machine type (`-8g`)?

Reply:
Yes I think we can! The naming seems consistent with H100(a3) and A100(a2). For a2, there are "a2-ultragpu-1g", "a2-ultragpu-2g", "a2-ultragpu-4g" and "a2-ultragpu-8g" and for a3 "a3-highgpu-8g". Other machine types can take the default value of "--accelerator=count=1" when the machine type is not ended with "-\dg". Will try this in this PR!

Reply:
Is this still something we'd want to do?

Reply:
Done!